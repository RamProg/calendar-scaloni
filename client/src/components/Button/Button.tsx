import { ButtonHTMLAttributes } from 'react';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  label: string;
  styleType?: 'primary' | 'secondary' | 'tertiary';
  styles?: string;
}

const Button: React.FC<ButtonProps> = ({
  label,
  styleType,
  styles,
  ...props
}) => {
  const getStyleString = (styleType: string) => {
    switch (styleType) {
      case 'primary':
      default:
        return 'bg-gray-200 hover:bg-gray-300 active:bg-gray-400 border-gray-500';
      case 'secondary':
        return 'bg-blue-200 hover:bg-blue-300 active:bg-blue-400 border-blue-500';
      case 'tertiary':
        return 'bg-red-200 hover:bg-red-300 active:bg-red-400 border-red-500';
    }
  };

  const styleString = getStyleString(styleType || 'primary');

  return (
    <button
      className={`p-2 rounded-lg font-sans font-medium transition-colors border ${styleString}} ${styles}`}
      {...props}
    >
      {label}
    </button>
  );
};

export default Button;
