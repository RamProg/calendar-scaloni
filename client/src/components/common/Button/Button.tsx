import { ButtonHTMLAttributes } from 'react';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  label: string;
  styleType?: 'primary' | 'secondary' | 'tertiary';
  big?: boolean;
  styles?: string;
}

const Button: React.FC<ButtonProps> = ({
  label,
  styleType,
  styles,
  big,
  ...props
}) => {
  const getStyleColor = (styleType: string | undefined) => {
    switch (styleType) {
      case 'tertiary':
        return 'red';
      case 'secondary':
        return 'blue';
      case 'primary':
      default:
        return 'gray';
    }
  };

  const styleColor = getStyleColor(styleType);

  return (
    <button
      className={`p-2 rounded-lg font-sans font-medium transition-colors border ${big && 'text-xl p-3'} ${styles} bg-${styleColor}-200 hover:bg-${styleColor}-300 active:bg-${styleColor}-400 border-${styleColor}-500`}
      {...props}
    >
      {label}
    </button>
  );
};

export default Button;
