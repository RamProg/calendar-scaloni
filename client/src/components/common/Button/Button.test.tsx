import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import Button from './Button.tsx';

describe('Button', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Button label="Test Button" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct label', () => {
    render(<Button label="Test Button" />);
    const buttonElement = screen.getByRole('button', { name: /Test Button/i });
    expect(buttonElement).toBeInTheDocument();
  });

  it('applies the correct styleType for secondary buttons', () => {
    render(<Button label="Test Button" styleType="primary" />);
    const buttonElement = screen.getByRole('button', { name: /Test Button/i });
    expect(buttonElement).toHaveClass(
      'bg-gray-200 hover:bg-gray-300 active:bg-gray-400 border-gray-500'
    );
  });

  it('applies the correct styleType for secondary buttons', () => {
    render(<Button label="Test Button" styleType="secondary" />);
    const buttonElement = screen.getByRole('button', { name: /Test Button/i });
    expect(buttonElement).toHaveClass(
      'bg-blue-200 hover:bg-blue-300 active:bg-blue-400 border-blue-500'
    );
  });

  it('applies the correct styleType for tertiary buttons', () => {
    render(<Button label="Test Button" styleType="tertiary" />);
    const buttonElement = screen.getByRole('button', { name: /Test Button/i });
    expect(buttonElement).toHaveClass(
      'bg-red-200 hover:bg-red-300 active:bg-red-400 border-red-500'
    );
  });

  test('calls onClick prop when clicked', () => {
    const handleClick = jest.fn();
    render(<Button label="Test Button" onClick={handleClick} />);
    fireEvent.click(screen.getByText(/test button/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
