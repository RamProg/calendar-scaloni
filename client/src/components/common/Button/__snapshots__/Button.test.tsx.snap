// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="p-2 rounded-lg font-sans font-medium transition-colors border undefined bg-gray-200 hover:bg-gray-300 active:bg-gray-400 border-gray-500 undefined"
  >
    Test Button
  </button>
</DocumentFragment>
`;
